src/commands/action/bite.ts

```diff
@@ -0,0 +1,106 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import {
+    ChatInputCommandInteraction, EmbedBuilder, GuildMember, SlashCommandBuilder
+} from "discord.js";
+import { Command } from "../../util/command-template.js";
+import { defaultErrorHandler } from "../../util/error-handler.js";
+import { TenorSingleton } from "../../util/tenor-utils.js";
+import { RED } from "../../util/colours.js";
+
+/*
+ * Creates an embed that states that the user had been bitten.
+ * The embed has an appropriate gif.
+ */
+
+const name: string = "bite";
+
+export const command: Command = {
+    // Command headers
+    data: <SlashCommandBuilder> new SlashCommandBuilder()
+        .setName(name)
+        .setDescription("Nom on someone!")
+        .setDMPermission(false)
+        .addUserOption(o =>
+            o.setName("target")
+                .setDescription("The user you want to bite")
+                .setRequired(true)
+        ),
+
+    // Command execution
+    async execute(ctx: ChatInputCommandInteraction): Promise<void> {
+        const tenor: TenorSingleton = TenorSingleton.getInstance();
+        const biter: GuildMember = <GuildMember> ctx.member;
+        const bit: GuildMember = <GuildMember> ctx.options.getMember("target") ?? null;
+        const gif: string = await tenor.getGifs("bite");
+
+        // If the user provided wasn't a member,
+        // tell the user so
+        if (!bit) {
+            await ctx.reply("Couldn't find that member... who are you trying to bite?");
+            return;
+        }
+
+        let response: string;
+
+        switch (true) {
+            case biter.id === bit.id:
+                // User tries to bonk themselves
+                response = "Wait, why would you do that?\n" +
+                           `${biter} bites themselves!`;
+                break;
+            case bit.id === ctx.client.user.id:
+                // User tries to bonk Sushi Bot
+                response = "Hey! I'm not *actually* edible\n" +
+                           `${ctx.client.user} bites ${biter}!!!`;
+                break;
+            default:
+                // User bonks someone else
+                response = `${biter} bites ${bit}. Yikes!`;
+                break;
+        }
+
+        const embed: EmbedBuilder = new EmbedBuilder()
+            .setDescription(response)
+            .setColor(RED)
+            .setImage(gif);
+
+        await ctx.reply({
+            embeds: [embed]
+        });
+    },
+
+    // Error handler
+    error: defaultErrorHandler,
+
+    // Help command embed
+    help: new EmbedBuilder()
+        .setTitle("Bite")
+        .setDescription("A bite command for when you feel peckish")
+        .setFields(
+            { name: "Format", value: `\`/${name} <target>\`` },
+            { name: "target", value: "Required parameter. The user you want to bite" }
+        )
+};
```

Comment:
This is the `/bite` command. It and the `/kiss` command are the new action commands that are similar to the existing action commands.